### ISSUE TITLE
Segmentation fault

### ISSUE THREAD

**Issue by RobinBoers** [oven-sh/bun#10992]
### How can we reproduce the crash?

I'm not sure if I can reproduce it. I'm building a multiplayer game using socket.io in TypeScript. The server suddenly crashed when I loaded the page in the browser. It never happened before.

### JavaScript/TypeScript code that reproduces the crash?

The project is here: https://github.com/dupunkto/oblikvo

### Relevant log output

```shell
Serving './public' on http://localhost:4000
Receiving disconnect
A client left the game.
============================================================
Bun v1.1.7 (b0b7db5c) macOS Silicon
Args: "bun", "--hot", "src/server/main.ts"
Features: jsc http_server transpiler_cache(7) tsconfig(8) 
Builtins: "bun:main" "node:async_hooks" "node:buffer" "node:crypto" "node:events" "node:fs" "node:http" "node:net" "node:path" "node:querystring" "node:stream" "node:string_decoder" "node:timers" "node:tty" "node:url" "node:util" "node:util/types" "node:zlib" "ws" 
Elapsed: 163701ms | User: 11545ms | Sys: 9803ms
RSS: 0.14GB | Peak: 0.14GB | Commit: 0.93GB | Faults: 66

panic(main thread): Segmentation fault at address 0x8
oh no: Bun has crashed. This indicates a bug in Bun, not your code.
```

### Stack Trace (bun.report)

Bun v1.1.7 ([`b0b7db5`](https://github.com/oven-sh/bun/tree/b0b7db5c060e830801b0273b7ff6e34e3eb5c254)) on macos aarch64 [AutoCommand]

Segmentation fault at address 0x00000008

- *2 unknown/js code*
- `JSC::InlineCacheCompiler::regenerate`
- `JSC::StructureStubInfo::addAccessCase`
- `JSC::repatchGetBy`
- `operationGetByIdOptimize`
- *3 unknown/js code*
- `llint_entry`

**Comment by RobinBoers** [comment]
It happened again:

<img width="1402" alt="image" src="https://github.com/oven-sh/bun/assets/60298132/0b9568fe-9f7e-4516-9312-24dbe63c2ac8">

It happened after refreshing the page in the browser (again).

**Comment by Polygons1** [comment]
try upgrading the Bun version maybe it will help

**Comment by Jarred-Sumner** [comment]
Yeah, this one should be fixed due to https://github.com/WebKit/WebKit/pull/28106 which is included in Bun v1.1.8
